### PR TITLE
COMMERCE-11644 Show exact dateTime for date field

### PR DESCRIPTION
**Pull request by pietrobocale**
https://issues.liferay.com/browse/COMMERCE-11644

### CODE REVIEW

**Comment by pietrobocale** [comment]
Base PR: #3843

**Comment by pietrobocale** [comment]
ci:test:sf

**Comment by pietrobocale** [comment]
ci:test:relevant

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
ci:stop

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase [hunk 1]
```diff
@@ -1612,6 +1612,144 @@ definition {
 		}
 	}
 
+	@description = "This is a test for the COMMERCE-11644."
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase:1615]: description is missing

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase [hunk 2]
```diff
@@ -1612,6 +1612,144 @@ definition {
...
+	@priority = 5
+	test CanProductMediaAndRelationShowCorrectDateFormat {
+		property portal.acceptance = "true";
+
+		var currentDate = DateUtil.getFormattedCurrentDate("yyyy", "GMT");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase:1620]: move this in a task

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase [hunk 3]
```diff
@@ -1612,6 +1612,144 @@ definition {
...
+
+			CommerceProducts.searchAndGoToProductEntry(entryProductName = "Simple Product");
+
+			CommerceJSONProductsAPI._addCommerceProduct(
+				catalogName = "Master",
+				productDescription = "Full Description",
+				productName = "Relation Product",
+				productType = "Simple",
+				purchasable = "true",
+				shortDescription = "Short Description");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase:1643]: Create the products using a for cycle

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase [hunk 4]
```diff
@@ -1612,6 +1612,144 @@ definition {
...
+				imageTitles = "Black",
+				validationImage = "true");
+		}
+
+		task ("Then the date of the new entry is using the correct format") {
+			for (var dateKeyWords : list "Seconds Ago,Minutes Ago,Hours Ago,Days Ago,Months Ago") {
+				AssertElementNotPresent(
+					key_columnNumber = 3,
+					key_productName = "Commerce_Black",
+					key_value = ${dateKeyWords},
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase:1659]: please move "Ago" from the list to the key in the assert

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase [hunk 5]
```diff
@@ -1612,6 +1612,144 @@ definition {
...
+					locator1 = "CommerceEntry#TABLE_SEARCH_BY_NAME_AND_COLUMN_CONTENT");
+			}
+
+			AssertElementPresent(
+				key_columnNumber = 3,
+				key_productName = "Commerce_Black",
+				key_value = ${currentDate},
+				locator1 = "CommerceEntry#TABLE_SEARCH_BY_NAME_AND_COLUMN_CONTENT");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase:1667]: please change the assertion to check the date format

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase [hunk 6]
```diff
@@ -1612,6 +1612,144 @@ definition {
...
+					key_value = ${dateKeyWords},
+					locator1 = "CommerceEntry#TABLE_SEARCH_BY_NAME_AND_COLUMN_CONTENT");
+			}
+
+			AssertElementPresent(
+				key_columnNumber = 3,
+				key_productName = "Commerce_White",
+				key_value = ${currentDate},
+				locator1 = "CommerceEntry#TABLE_SEARCH_BY_NAME_AND_COLUMN_CONTENT");
+		}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase:1688]: same as above

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase [hunk 7]
```diff
@@ -1612,6 +1612,144 @@ definition {
...
+					key_value = ${dateKeyWords},
+					locator1 = "CommerceEntry#TABLE_SEARCH_BY_NAME_AND_COLUMN_CONTENT");
+			}
+
+			AssertElementPresent(
+				key_columnNumber = 4,
+				key_productName = "Relation Product",
+				key_value = ${currentDate},
+				locator1 = "CommerceEntry#TABLE_SEARCH_BY_NAME_AND_COLUMN_CONTENT");
+		}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase:1750]: same as above

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

**Comment by pietrobocale** [comment]
ci:test:sf

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
ci:test:relevant

**Comment by pietrobocale** [comment]
ci:test:sf

**Comment by pietrobocale** [comment]
ci:test:sf

**Comment by pietrobocale** [comment]
ci:test:relevant

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
ci:test:relevant

**Comment by pietrobocale** [comment]
The failures are not related to the PR

**Comment by luca-pellizzon** [comment]
ci:forward:force

**Pull request closed without merging**